CMSgov/onemac
# Oy2 23387 - display last rai response date in package table and details page for CMS users

Story: https://qmacbis.atlassian.net/browse/OY2-23387
Endpoint: See github-actions bot comment

### Details

Add "Formal RAI Received" column to package view and as a field on detail view. This is only visible by CMS users.

### Changes

- Add column to Waiver Package Table for CMS Users
- Add field to details page for CMS Users
- Updated buildAnyPackage to set latestRaiResponseTimestamp when an RAI is submitted
- Updated migrate.js to update all existing packages to have new latestRaiResponseTimestamp field
- reworked how waiver detail default fields were configured

### Implementation Notes

- Added new property to DetailAttribute that allows specification of a UserRole property (privilege) by which the currently logged in user must have in order for the field to be displayed

### Test Plan

1. Login as CMS user
2. Verify "Formal RAI Response" exists in package view on waivers tab as a column
3. Locate package that has an RAI Response and verify value in column matches latest RAI Response date
4. Navigate to Details page for a package with an RAI Response and verify value of "Formal RAI Response" matches latest RAI Respsonse date
5. Login as State user
6. Verify column does not exist in package view on waivers tab or on details page
7. Locate a package in RAI Issued status and submit an RAI Response
8. Login as CMS User and verify current date appears in "Formal RAI Response" column and field
